lib/prelude/input.dart

```diff
@@ -1,14 +1,216 @@
+      if (tapInteractions.containsKey(tap.pointerId)) {
+        result.add(tapInteractions[tap.pointerId]!);
+      }
+    }
+    for (final longTap in longTaps) {
+      if (tapInteractions.containsKey(longTap.pointerId)) {
+        result.add(tapInteractions[longTap.pointerId]!);
+      }
+    }
+    return result;
+  }
+
+  /// Get the taps and long taps which ended this frame
+  List<TapInteraction> newTapUps() {
+    final result = <TapInteraction>[];
+    for (final tapUp in tapUps) {
+      if (tapInteractions.containsKey(tapUp.pointerId)) {
+        result.add(tapInteractions[tapUp.pointerId]!);
+      }
+    }
+    return result;
+  }
+
+  /// Get the taps and long taps which were cancelled this frame
+  List<TapInteraction> newTapCancels() {
+    final result = <TapInteraction>[];
+    for (final tapCancel in tapCancels) {
+      if (tapInteractions.containsKey(tapCancel.pointerId)) {
+        result.add(tapInteractions[tapCancel.pointerId]!);
+      }
+    }
+    return result;
+  }
+
+  /// Get the drags which started this frame
+  List<DragInteraction> newDragStarts() {
+    final result = <DragInteraction>[];
+    for (final dragStart in dragStarts) {
+      if (dragInteractions.containsKey(dragStart.pointerId)) {
+        result.add(dragInteractions[dragStart.pointerId]!);
+      }
+    }
+    return result;
+  }
+
+  /// Get the drags which updated this frame
+  List<DragInteraction> newDragUpdates() {
+    final result = <DragInteraction>[];
+    for (final dragUpdate in dragUpdates) {
+      if (dragInteractions.containsKey(dragUpdate.pointerId)) {
+        result.add(dragInteractions[dragUpdate.pointerId]!);
+      }
+    }
+    return result;
+  }
+
+  /// Get the drags which ended this frame
+  List<DragInteraction> newDragEnds() {
+    final result = <DragInteraction>[];
+    for (final dragEnd in dragEnds) {
+      if (dragInteractions.containsKey(dragEnd.pointerId)) {
+        result.add(dragInteractions[dragEnd.pointerId]!);
+      }
+    }
+    return result;
+  }
+
+  void clear() {
+    keysPressed = {};
+    taps = [];
+    longTaps = [];
+    tapUps = [];
+    tapCancels = [];
+    dragStarts = [];
+    dragUpdates = [];
+    dragEnds = [];
+  }
+
+  void process(double dt) {
+    // Set last frame position for drags
+    for (final interaction in dragInteractions.values) {
+      interaction.positionAtLastFrame = interaction.position;
+    }
+
+    // Use the events from this frame to update the pointer interactions
+    // -- normal taps
+    for (final tap in taps) {
+      assert(tapInteractions.containsKey(tap.pointerId) == false,
+          'Pointer ID ${tap.pointerId} already exists');
+
+      tapInteractions[tap.pointerId] = TapInteraction(
+        tapDown: tap,
+        long: false,
+      );
+    }
+    // -- long taps
+    for (final longTap in longTaps) {
+      assert(tapInteractions.containsKey(longTap.pointerId) == false,
+          'Pointer ID ${longTap.pointerId} already exists');
```

Comment:
Same as above, pasta copy?

Reply:
This has the same intention behind it, long taps to the same thing as taps, just with the long flag.